url/src/lib.rs

```diff
@@ -3043,3 +3046,89 @@ impl<'a> Drop for UrlQuery<'a> {
         }
     }
 }
+
+#[cfg(all(feature = "uniffi", test))]
+uniffi::setup_scaffolding!();
+
+#[cfg(feature = "uniffi")]
+mod uniffi_core_impl {
+    use std::str::FromStr;
+
+    use super::*;
+
+    use uniffi_core::{
+        deps::bytes::{Buf, BufMut},
+        metadata, FfiConverter, MetadataBuffer,
+    };
+
+    uniffi_core::derive_ffi_traits!(blanket Url);
+
+    unsafe impl<UT> FfiConverter<UT> for Url {
+        uniffi_core::ffi_converter_rust_buffer_lift_and_lower!(UT);
+
+        fn write(obj: Self, buf: &mut Vec<u8>) {
+            let string: String = obj.to_string();
+            <String as FfiConverter<UT>>::write(string, buf)
+        }
+
+        fn try_read(buf: &mut &[u8]) -> uniffi_core::Result<Self> {
+            <String as FfiConverter<UT>>::try_read(buf)
+                .and_then(|s| Url::from_str(s.as_str()).map_err(|e| e.into()))
+        }
+
+        const TYPE_ID_META: MetadataBuffer =
+            MetadataBuffer::from_code(metadata::codes::TYPE_STRING);
```

Comment:
I'm also a bit unsure about the implications of reusing `TYPE_STRING` as `TYPE_ID_META`, possibly more correct to use `::TYPE_CUSTOM`.